roles/kubernetes/kubeadm/tasks/main.yml

```diff
@@ -30,7 +30,10 @@
   run_once: true
 
 - name: Calculate kubeadm CA cert hash
-  shell: set -o pipefail && openssl x509 -pubkey -in {{ kube_cert_dir }}/ca.crt | openssl rsa -pubin -outform der 2>/dev/null | openssl dgst -sha256 -hex | sed 's/^.* //'
+  shell: |
+    set -o pipefail && openssl x509 -pubkey -in {{ kube_cert_dir }}/ca.crt | \
+    openssl {% if 'RSA' in kube_asymmetric_encryption_algorithm %}rsa{% elif 'ECDSA' in kube_asymmetric_encryption_algorithm %}ec{% else %}rsa{% endif %} -pubin -outform der 2>/dev/null | \
+    openssl dgst -sha256 -hex | sed 's/^.* //'
```

Comment:
Ouch. We really need to switch to community.crypto at some point ^.

Reply:
😊 I tried to optimize this hash calculation logic. https://github.com/kubernetes-sigs/kubespray/pull/11758